File: src/LiWiMus.Web/Profiles/UserProfile.cs
using AutoMapper;
using LiWiMus.Core.Entities;
using LiWiMus.Web.Areas.User.ViewModels;

namespace LiWiMus.Web.Profiles;

// ReSharper disable once UnusedType.Global
public class UserProfile : Profile
{
    public UserProfile()
    {
        CreateMap<ProfileViewModel, User>()
            .ForMember(user => user.Gender, opt => opt.MapFrom(vm => vm.IsMale ? Gender.Male : Gender.Female))
            .ReverseMap()
            .ForPath(vm => vm.IsMale, opt => opt.MapFrom(user => user.Gender == Gender.Male));
    }
}

File: src/LiWiMus.API/Areas/User/Profiles/User.Profile.fs
namespace LiWiMus.API.Areas.User.Profiles

open AutoMapper
open LiWiMus.API.Areas.User.Dto
open LiWiMus.Core.Users
open Microsoft.FSharp.Core
open LiWiMus.API.Extensions

type UserProfile() as this =
    inherit Profile()
    do this.CreateMap<User, UserDtoReadOne>()
           .ForMemberFs((fun dto -> dto.Email), fun opt -> opt.MapFromFs(fun e -> e.Balance))
           .ReverseMap()
       |> ignore
    
    do this.CreateMap<User, UserDtoCreate>()
            .ReverseMap()
       |> ignore